nrf52-hal-common/src/spim.rs

```diff
@@ -29,8 +27,71 @@ use embedded_hal::digital::v2::OutputPin;
 /// - The SPIM instances share the same address space with instances of SPIS,
 ///   SPI, TWIM, TWIS, and TWI. You need to make sure that conflicting instances
 ///   are disabled before using `Spim`. See product specification, section 15.2.
+#[derive(Debug)]
 pub struct Spim<T>(T);
 
+/// An ongoing SPI transaction that is holding the CS pin low.
+#[derive(Debug)]
+pub struct SpiTransaction<'a, T> {
+    chip_select: &'a mut Pin<Output<PushPull>>,
+    spim: &'a mut T,
+}
+
+/// An ongoing transfer that was initiated by a call to `.transfer_polling()`.
+///
+/// This transfer must be polled to completion.  Failing to poll it until completed might leave the
+/// peripheral in an inconsistent state.
+#[derive(Debug)]
+#[must_use = "This transfer must be polled to completion.  Failing to poll it until completed might leave the peripheral in an inconsistent state."]
+pub struct SpiTransfer<'a, T> {
+    spim: &'a mut T,
+    state: TransferState,
+    chunks: core::slice::ChunksMut<'a, u8>,
+}
+
+/// An ongoing transfer that was initiated by a call to `.transfer_even_polling()`.
+///
+/// This transfer must be polled to completion.  Failing to poll it until completed might leave the
+/// peripheral in an inconsistent state.
+#[derive(Debug)]
+#[must_use = "This transfer must be polled to completion.  Failing to poll it until completed might leave the peripheral in an inconsistent state."]
+pub struct SpiEvenTransfer<'a, T> {
+    spim: &'a mut T,
+    state: TransferState,
+    chunks: core::iter::Zip<core::slice::Chunks<'a, u8>, core::slice::ChunksMut<'a, u8>>,
```

Comment:
It would be easier to read if you use a type alias for this (and the other iterators)

Reply:
Hmm I couldn't find a good name for these...  In this particular case, for example, we could alias `Chunks` as `SentChunks` or something, but we would then still need the `Zip` type.  I also didn't find a good alias for `ChunksMut` because in some cases, it will be the `ReceivedChunks`, while in some cases, it will be both sent and received chunks...

Reply:
I will at least simplify the imports a bit!